src/dotnet/APIView/APIViewWeb/wwwroot/js/site.js

```diff
@@ -4,10 +4,52 @@
 // Write your Javascript code.
 $(function () {
     $(".commentable").click(function () {
-        var myForm = $("#commentForm");
-        $("#idBox").val(this.id);
-        myForm.show();
-        $("#commentThread").focus();
+        var formExists = $($(this).parents(".code-line")[0].nextElementSibling).find("#comment-form").length > 0;
+
+        if (!formExists) {
+            let myForm = $("#comment-form");
+            let clone = myForm.clone();
+            clone[0].querySelector("#id-box").setAttribute("value", this.id);
+            let stringRep = clone[0].outerHTML;
+
+            var thread = $($(this).parents(".code-line")[0].nextElementSibling).find(".comment-thread-contents");
+            if (thread.length > 0) {
+                thread.after(stringRep);
+                $($(this).parents(".code-line")[0].nextElementSibling).find(".review-thread-reply").hide();
+            }
+            else {
+                $(this).parents(".code-line").after("<tr><td>" + stringRep + "</td></tr>");
+            }
+        }
+        $($(this).parents(".code-line")[0].nextElementSibling).find(".comment").show();
```

Comment:
https://api.jquery.com/first/ instead of `[0]`, .next() instead of `nextElementSibling`